hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/NodeInfoImpl.java

```diff
@@ -20,29 +20,45 @@
 import static java.util.Objects.requireNonNull;
 
 import com.hedera.hapi.node.base.AccountID;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
 import com.swirlds.common.utility.CommonUtils;
 import com.swirlds.platform.system.address.Address;
 import com.swirlds.state.spi.info.NodeInfo;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.security.cert.CertificateEncodingException;
 
 public record NodeInfoImpl(
         long nodeId,
         @NonNull AccountID accountId,
         long stake,
         @NonNull String externalHostName,
         int externalPort,
+        @NonNull String internalHostName,
+        int internalPort,
         @NonNull String hexEncodedPublicKey,
-        @NonNull String memo)
+        @NonNull String memo,
+        @Nullable Bytes sigCertBytes)
         implements NodeInfo {
     @NonNull
     static NodeInfo fromAddress(@NonNull final Address address) {
+        final var sigCert = address.getSigCert();
+        Bytes sigCertBytes;
+        try {
+            sigCertBytes = sigCert == null ? null : Bytes.wrap(sigCert.getEncoded());
+        } catch (CertificateEncodingException e) {
+            sigCertBytes = null;
```

Comment:
Instead of null can we keep `Bytes.EMPTY`

Reply:
Changed.